src/gleam/int.gleam

```diff
@@ -515,20 +515,39 @@ fn do_undigits(
 }
 
 /// Generates a random int between the given minimum and maximum values.
-///
+/// if `inclusive < exclusive`, returns a value in `[inclusive, exclusive)`
+/// if `exclusive < inclusive`, returns a value in `(exclusive, inclusive]`
+/// if `inclusive == exclusive`, always returns `inclusive`
+/// 
 /// ## Examples
 ///
 /// ```gleam
 /// > random(1, 5)
 /// 2
 /// ```
-///
-pub fn random(min: Int, max: Int) -> Int {
-  float.random(to_float(min), to_float(max))
-  |> float.floor()
-  |> float.round()
+pub fn random(inclusive: Int, exclusive: Int) -> Int {
+  do_random(inclusive, exclusive)
+}
+
+@target(erlang)
+fn do_random(inclusive: Int, exclusive: Int) -> Int {
+  case inclusive, exclusive {
+    _, _ if inclusive == exclusive -> inclusive
+    max, min if inclusive > exclusive -> max - do_random_uniform(max - min)
+    min, max if inclusive < exclusive -> max - do_random_uniform(max - min)
```

Comment:
The compiler yells at me here when I build with javascript target that `do_random_uniform` does not have an implementation, even though it's only used in a function that's for the erlang target.

Reply:
@target is discouraged, new version just uses regular external functions